### PR TITLE
Export some of the functionality in `xray::generation`

### PR DESCRIPTION
**Pull request by krzesi-mir**

### CODE REVIEW

File: quadtree/src/lib.rs [hunk 1]
```diff
@@ -217,9 +217,10 @@ impl FromStr for NodeId {
     }
 }
 
+pub const NODE_PREFIX: char = 'r';
 impl fmt::Display for NodeId {
     fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
-        formatter.write_char('r')?;
+        formatter.write_char(NODE_PREFIX)?;
```
Comment (feuerste) [inline quadtree/src/lib.rs]: What do you think about changing this to
```rust
pub const NODE_PREFIX: &str = "r";
...
        formatter.write_str(NODE_PREFIX)?;
```
then we wouldn't need `&NODE_PREFIX.to_string()` in `utils.rs`?

File: xray/src/utils.rs [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
 
+const META_PREFIX: &str = "meta";
+
 pub fn get_meta_pb_path(directory: &Path, id: NodeId) -> PathBuf {
-    directory.join(&format!("{}.pb", id).replace("r", "meta"))
+    directory.join(&format!("{}.pb", id).replace(&NODE_PREFIX.to_string(), META_PREFIX))
+}
+
+pub fn get_root_node_id_from_meta_pb_path(meta_path: &Path) -> io::Result<NodeId> {
+    let error_message = format!("Invalid path {:?}.", meta_path);
```
Comment (feuerste) [inline xray/src/utils.rs]: How about changing this into
```rust
let invalid_input_error = || {
    io::Error::new(io::ErrorKind::InvalidInput, format!("Invalid path {:?}.", meta_path))
};
```
and calling `invalid_input_error()` below twice?

File: quadtree/src/lib.rs (hunk 1 above)
Comment (krzesi-mir) [reply quadtree/src/lib.rs:223]: done :+1:

File: xray/src/utils.rs (hunk 2 above)
Comment (krzesi-mir) [reply xray/src/utils.rs:13]: Good idea! changed.

**Comment by krzesi-mir** [comment]
Thanks for the review, Marco!

**Pull request merged**